gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientWriteChannel.java

```diff
@@ -131,7 +132,7 @@ public Boolean call() throws Exception {
           }
           byteBuffer.put(data.toByteArray());
           // switch to read mode
-          byteBuffer.flip();
+          ((Buffer) byteBuffer).flip();
```

Comment:
Why we are doing this? What will happen if we do not do this?

Also, add a comment here.

Reply:
It is covered as part of PR description.